### ISSUE TITLE
Runtime disable for connecting to DB

### ISSUE THREAD

**Issue by jakeleventhal** [prisma/prisma#21126]
## Problem

```ts
const client = (datasourceUrl: string | undefined) => new PrismaClient({ datasourceUrl });

// Export the prisma client
export const getPrismaClient = (url?: string): PrismaClient => {
  // @ts-ignore
  if (typeof window === 'undefined') {
    return client(url);
  }

  // @ts-ignore
  return null;
};

const prisma = getPrismaClient();
export default prisma;
export * from '@prisma/client;
```

I do this in a shared pnpm workspace package so that if i use this package in the backend, i can import the prisma client just fine. But if i want to import a type in the frontend, like a User model, it doesn't try to connect to the DB in the browser. Is there a better way to do this? Shutting of DB connection attempts via some runtime override would simplify this code and make it such that i dont have to do `// @ts-ignore` to lie to TS

## Suggested solution

A runtime disable for connecting to DB

## Alternatives

See example

**Comment by janpio** [comment]
So you want to import the file above both in backend and browser, and get rig of the `@ts-ignore` that you needed to use to be able to do the `if ... return null` that makes it possible to use it both in backend and browser?

**Comment by jakeleventhal** [comment]
Yes, more context is that I'm using a PNPM monorepo with several applications. Each application has its own schema and thus its own package "`@<app>/database`" that looks like the above file.

This allows me easily access types/client in each application like so:
```ts
// in app1
import prisma, { User } from '@app1/datatbase'
await prisma.user... // <- makes connections to the app1 database

// in app2
import prisma, { DifferentUser } from '@app1/datatbase'
await prisma.differentUser... // <- makes connections to the app2 database
```

**Comment by SevInf** [comment]
@jakeleventhal sorry, I don't quite understand what is the issue you are experiencing. Types from prisma client are imprtable to the frontend without instantiating client:

```
import type { User } from '@prisma/client';
```

If you are talking about accessing `prisma.user` object in the frontend - no method in there works without DB connection.

If your monorepo packaes entry-point contain both client instantiation and types re-export and that's the problem - you can split types re-exports into it's own file:

```ts
import type { User } from '@app1/database/types`
```

**Comment by jakeleventhal** [comment]
@SevInf this solution doesn't quite work. What if i want to import enums? i 
<img width="1085" alt="Screenshot 2023-09-29 at 7 35 21 AM" src="https://github.com/prisma/prisma/assets/15133828/926fc0aa-eccd-4bb2-b2cf-75769f3b2d21">

**Comment by SevInf** [comment]
@jakeleventhal You can import enums into your frontend bundle without instantiating the client, just skip `type` keyword.

**Comment by jakeleventhal** [comment]
Yeah but then we are back to the same place where I need the same code from the original post. It will try to instantiate the client unless I use that special logic to conditionally instantiate my client

**Comment by jakeleventhal** [comment]
how would that solution work with autocomplete? if there is no `app1/database/index.ts` then trying to use auto-import in a monorepo wouldnt be able to have any knowledge of the module exports

**Comment by jakeleventhal** [comment]
I'm not sure, I am no longer using prisma though so feel free to leave this issue open, or close it.
